src/Rules/Functions/RandomIntParametersRule.php

```diff
@@ -59,13 +63,18 @@ public function processNode(Node $node, Scope $scope): array
 			if (!$maxPermittedType->isSuperTypeOf($maxType)->yes()) {
 				$message = 'Parameter #1 $min (%s) of function random_int expects lower number than parameter #2 $max (%s).';
 
-				return [
-					RuleErrorBuilder::message(sprintf(
-						$message,
-						$minType->describe(VerbosityLevel::value()),
-						$maxType->describe(VerbosityLevel::value())
-					))->build(),
-				];
+				// True if sometimes the parameters conflict.
+				$isMaybe = !$maxType->isSuperTypeOf($minType)->no();
+
+				if (!$isMaybe || $this->reportMaybes) {
```

Comment:
This needs a level integration test. Add some reproducing code to [`acceptTypes.php`](https://github.com/phpstan/phpstan-src/blob/master/tests/PHPStan/Levels/data/acceptTypes.php). Run [LevelsIntegrationTest](https://github.com/phpstan/phpstan-src/blob/master/tests/PHPStan/Levels/LevelsIntegrationTest.php) and it will automatically update the level JSON files with the first occurence of the error. Levels 5 and 7 should be updated.

Reply:
And done.